lib/resty/socket.lua

```diff
@@ -56,12 +68,13 @@ do
       local ssl = require 'ssl'
       local params = {
         mode = 'client',
-        protocol = 'tlsv1',
+        protocol = 'any',
```

Comment:
For this fix we'd want to introduce `options.protocol`, the patch should be identical with the one in lua-cassandra: https://github.com/thibaultcha/lua-cassandra/pull/141

Reply:
shouldn't we leave this at `any` and then make the `options` below configurable? That would allow full control to the user, but leave room for flexibility, and be secure by excluding the bad ones as a default?

Reply:
added